lib/WP_Auth0_LoginManager.php

```diff
@@ -587,6 +576,68 @@ public function end_session() {
 		}
 	}
 
+	/**
+	 * Get and filter the scope used for access and ID tokens
+	 *
+	 * @param string $context - how are the scopes being used?
+	 *
+	 * @return string
+	 */
+	public static function get_userinfo_scope( $context = '' ) {
+		return apply_filters( 'auth0_auth_token_scope', 'openid email name nickname picture', $context );
+	}
+
+	/**
+	 * Get a base authorize URL for handling HLP redirects.
+	 *
+	 * @param null|string $connection - specific connection to use, pass null to use all enabled.
+	 * @param null|string $redirect_to - URL to redirect upon successful authentication.
+	 *
+	 * @return array
+	 */
+	public static function get_authorize_params( $connection = null, $redirect_to = null ) {
+		$params = array();
+		$options = WP_Auth0_Options::Instance();
+		$lock_options = new WP_Auth0_Lock10_Options();
+		$is_implicit = (bool) $options->get( 'auth0_implicit_workflow', FALSE );
+		$nonce = WP_Auth0_Nonce_Handler::getInstance()->get();
+
+		$params[ 'client_id' ] = $options->get( 'client_id' );
+		$params[ 'scope' ] = self::get_userinfo_scope( 'authorize_url' );
+		$params[ 'response_type' ] = $is_implicit ? 'id_token': 'code';
+		$params[ 'redirect_uri' ] = $is_implicit
+			? $lock_options->get_implicit_callback_url()
+			: $options->get_wp_auth0_url( null, $is_implicit );
+
+		if ( $is_implicit ) {
+			$params[ 'nonce' ] = $nonce;
+		}
+
+		if ( ! empty( $connection ) ) {
+			$params[ 'connection' ] = $connection;
+		}
+
+		// Get the telemetry header.
+		$telemetry = WP_Auth0_Api_Client::get_info_headers();
+		$params[ 'auth0Client' ] = $telemetry[ 'Auth0-Client' ];
+
+		// Where should the user be redirected after logging in?
+		if ( empty( $redirect_to ) && ! empty( $_GET['redirect_to'] ) ) {
+			$redirect_to = $_GET['redirect_to'];
+		} elseif ( empty( $redirect_to ) ) {
+			$redirect_to = $options->get( 'default_login_redirection' );
+		}
+
+		// State parameter, checked during login callback.
+		$params[ 'state' ] = base64_encode( json_encode( array(
+			'interim' => false,
+			'nonce' => $nonce,
+			'redirect_to' => filter_var( $redirect_to, FILTER_SANITIZE_URL ),
```

Comment:
why do you need to manipulate the `redirect_to` value? can't you use directly the one defined just a few lines above this?

Reply:
Since this could come from a URL param or a settings field, I want to make sure that it's an actual URL.

Reply:
and why you don't bother about that in the lines above?

Reply:
Can you clarify what your question is here? Do what to which lines?

Reply:
Line 628 or 630 define what `$redirect_to` is going to be. Then in this line (637) you set as the `redirect_to` property (inside the state object) `filter_var( $redirect_to, FILTER_SANITIZE_URL )` as value. What's the difference and why do you need to call `filter_var` first this time?

Reply:
Line 628 and 630 pull a raw value from the URL or from the database, could be anything. So if a redirect URL is not passed and there's one in the URL, get that. Otherwise if it's not passed but nothing in the URL, get the default. Then take that value, get rid of non-URL-y characters, and pack it up into `state` to be sent to Auth0 and returned. If that's null/empty in the callback then it will go to the homepage.

Hopefully that answers your question? If not, ask another way and I'll try again 😀